### PR TITLE
Show github upload progress in the leetcode when submitted

### PR DESCRIPTION
**Pull request by sbibek**
LeetHub currently doesn't report any kind of progress when submission is being uploaded to GitHub. The only way to check if LeetHub uploaded the submission is to check if the submission count in the extension popup has increased or by checking the GitHub repo itself. This PR adds the functionality of showing upload progress directly in the Leetcode. When the submission is successful and the LeetHub upload is triggered, a progress spinner on the left side of the "Run Code" button pops up. If the upload is successful, the spinner is replaced by a green check. Currently, the LeetHub codebase doesn't have a proper way to pass success/failure of the upload calls, I have added a callback for success just for this purpose. Adding one for failure seems like a bit of work right now. So, this feature starts a timeout of 10 seconds (should be sufficient) after the upload has started. If the upload is not completed by the end of 10 seconds, then the upload is concluded to have failed and a red check replaces the spinner. 

Addresses #112 

progresspr

### CODE REVIEW

File: scripts/leetcode.js [hunk 1]
```diff
@@ -520,12 +538,65 @@ const loader = setInterval(() => {
...
+          }
         ); // Encode `code` to base64
       }, 1000);
     }
   }
 }, 1000);
 
+/* Since we dont yet have callbacks/promises that helps to find out if things went bad */
+/* we will start 10 seconds counter and even after that upload is not complete, then we conclude its failed */
+function startUploadCountDown() {
```
Comment (QasimWani) [inline scripts/leetcode.js:556]: i like the overall idea, but 10seconds is a lot. Currently, we push the code when all test cases are passed. To check for that, we simply check for "Success" in the top left corner of the page. 
What if u instead create an event listener whenever we pressed the submit button and look for [success tag](https://github.com/QasimWani/LeetHub/blob/main/scripts/leetcode.js#L397). Once that condition is true, and whle it's submitting u can keep the spinner running.
But the idea is that as soon as someone hits submit, the spinner is activated.

**Review by QasimWani — CHANGES_REQUESTED** [review]
please see comments and change structure accordingly.
Would love to see this in production!

File: scripts/leetcode.js (hunk 1 above)
Comment (sbibek) [reply scripts/leetcode.js:556]: The spinner is activated by the success tag. The spinner will shut down and show a green check when code upload is successful. The 10s countdown is just for the case when the upload is unsuccessful. It's just if the code is not uploaded within 10s, then there probably was an error. But if there are no errors, spinners go away like in 3/4 seconds when the upload is complete. Does this make sense?

Comment (sbibek) [reply scripts/leetcode.js:556]: 10s thing is just an edge case handler when pushing to the Github repo fails. For example, if there are any issues with the OAuth tokens. Please let me know if that makes sense.

**Comment by sbibek** [comment]
Like you suggested, the spinner activates on the following condition
_if the submission was successful and the "success" tag appears on the top._
A green check replaces the spinner as soon as upload to Github is successful.

The 10s function is there to just handle an edge case if the upload is unsuccessful for any reason. If the upload to Github is successful, the 10s counter is shut down. But if the upload is unsuccessful and the counter reaches 10s, the upload is tagged unsuccessful. 

Does this make sense?

**Comment by SysSn13** [comment]
These changes are breaking the flow of LC Explore page submissions. Please add support for explore section or handle the errors properly. [This line ](https://github.com/QasimWani/LeetHub/blob/main/scripts/leetcode.js#L614) is causing error in one case. There might be more cases.

**Comment by sbibek** [comment]
@SysSn13 will check that.

**Comment by sbibek** [comment]
@SysSn13 , I added the support for the "explore section" and a generic error handler is in place so that it doesn't break any other features. Any other areas you can think of this might affect?

**Comment by SysSn13** [comment]
@sbibek  I think that's enough. For leetcode, there are only two types of submission right now, normal problem and explore section problem. Make sure both of them are working.

**Comment by sbibek** [comment]
@SysSn13 , Thanks.

**Comment by QasimWani** [comment]
have all the issues been fixed @sbibek? Made the cardinal sin of approving a PR without looking at it.
@SysSn13 assigned you to take charge of this PR.

**Comment by SysSn13** [comment]
#120  A generic error handler is introduced for it which resolves this issue.

**Comment by sbibek** [comment]
@QasimWani @SysSn13, Explore Submissions and cardinality checks available on #120. I think maybe we can create some kind of checklist of features so that we can verify the new features against that?

**Comment by vrindavan** [comment]
My status is just red for a while, I have tried installing again but none works. Also tried to disconnect from GitHub Account settings and re-connect but the same issue.
image

**Pull request merged**